### ISSUE TITLE
moreh_softmax_h_large fails with a TRISC0 watcher assert

### ISSUE THREAD

**Issue by iwroszTT** [tenstorrent/tt-metal#53927]
### Description

  ttnn.softmax with dim=-2 and H large enough to select SoftmaxProgramFactoryGeneralHLarge trips a device-side assert on TRISC0 whenever the watcher is enabled.

```
  TT_THROW: Device 0: Aborting wait due to watcher error: Device 0 worker core(x= 0,y= 0):             
  TRISC0 tripped an assert on line 262. Note that file name reporting is not yet implemented, and the reported line number for the assert may be from a different file. Current kernel: ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/kernels/moreh_softmax_h_large.cpp.
```

###  Repro (any watcher-enabled build):

  TT_METAL_WATCHER=5 python -c "
  import torch, ttnn
  dev = ttnn.open_device(device_id=0)
  x = ttnn.from_torch(torch.zeros(1, 1, 4096, 32), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,  device=dev)                                       
  ttnn.softmax(x, dim=-2)"

  H=4096 (Ht=128) overflows the 512KB small-path budget in is_softmax_general_h_small_available, forcing the H-large factory.

### Observations                                      
                                                    
  - Deterministic on wh_n150_civ2 and bh_p150b_civ2_viommu (merge-gate run 32476721902, PR #53677) and locally on BH p100a (reported assert line 258 there — the line number is build-dependent and likely from an included header).
  - Without the watcher the kernel produces correct results (exact uniform softmax golden passes), so the defect is latent in normal runs — plausibly the forbidden-CB-access class the recent watcher machinery detects (#39432-adjacent), on the unpack thread.
  - The W-large sibling (moreh_softmax_w_large.cpp, reached via rank-5 dim=-1 at W=4096) is clean under the watcher on both SKUs — the defect is specific to the H-large kernel.
  - Never caught before because no gating test exercised this factory and the nightly moreh softmax tests run without the watcher.

  Test hook: PR #53677 keeps a merge-gate cell for this factory as a disabled repro (NormalizationSmoke.DISABLED_SoftmaxGeneralHLarge in tests/ttnn/unit_tests/gtests/test_normalization.cpp). When fixing, please remove the DISABLED_ prefix so the factory becomes merge-gate-covered.
